README.md

```diff
@@ -195,52 +195,75 @@ Install custom reporters as a development dependency and you can use the `--repo
 
-Globs are matched using [minimatch](https://www.npmjs.com/package/minimatch).
+Nyc will instrument all files if the `--all` flag is set.
+In this case all files will appear in the coverage report and contribute to coverage statistics.
 
-For example, the following config will exclude any files with the extension `.spec.js`,
-and anything in the `build` directory:
+Nyc will only cover files that are located under `cwd`, and then only `*.js` files or files with extensions listed in in the `nyc.extension` array.
 
-```json
-{
-  "nyc": {
-    "exclude": [
-      "**/*.spec.js",
-      "build"
-    ]
-  }
-}
-```
-> Note: Since version 9.0 files under `node_modules/` are excluded by default.
-  add the exclude rule `!**/node_modules/` to stop this.
+You can reduce the set of covered files by adding `include` and `exclude` filter arrays to your config.
+These allow you to shape the set of covered files by specifying glob patterns that can filter files from the covered set.
+The `exclude` array may also use exclude negated glob patterns, these are specified with a `!` prefix, and can restore sub-paths of excluded paths.
+
+Globs are matched using [minimatch](https://www.npmjs.com/package/minimatch).
 
-> Note: exclude defaults to `['coverage/**', 'test/**', 'test{,-*}.js', '**/*.test.js', '**/__tests__/**', '**/node_modules/**']`,
-which would exclude `test`/`__tests__` directories as well as `test.js`, `*.test.js`,
-and `test-*.js` files. Specifying your own exclude property overrides these defaults.
+We use the following process to remove files from consideration,
+ * First, limit the set of covered files to those files in paths listed in the `include` array.
+ * Then, remove any files that are found in the `exclude` array.
+ * Finally, restore any exclude negated files that have been excluded in the second step
+
+
+### Using include and exclude arrays
+
+If there are paths specified in the `include` array, then the set of covered files will be limited to eligible files found in those paths.
+If the `include` array is left undefined all eligible files will be included, equivalent to setting `include: ['**']`.
+Include options can be specified on the command line with the `-n` switch.
+
+If there are paths specified in the `exclude` array, then the set of covered files will not feature eligible files found in those paths.
+You can also specify negated paths in the `exclude` array, by prefixing them with a `!`.
+Negated paths can restore paths that have been already been excluded in the `exclude` array.
+Exclude options can be specified on the command line with the `-x` switch.
+
+`Exclude` has the following defaults settings:
+```js
+[
+  'coverage/**',
+  'test/**',
+  'test{,-*}.js',
+  '**/*.test.js',
+  '**/__tests__/**',
+  '**/node_modules/**'
+]
+```
+These settings exclude `test`/`__tests__` directories as well as `test.js`, `*.test.js`, and `test-*.js` files.
```

Comment:
Lets say "`test` and `__tests__` directories" to avoid using the slash between two folders.

Also the default list is a bit outdated.  From [test-exclude](https://github.com/istanbuljs/istanbuljs/blob/master/packages/test-exclude/index.js):
```js
exportFunc.defaultExclude = [
    'coverage/**',
    'packages/*/test/**',
    'test/**',
    'test{,-*}.js',
    '**/*{.,-}test.js',
    '**/__tests__/**',
    '**/node_modules/**',
    '**/babel.config.js'
];
```

Reply:
Oh, I didn't check that, that was slight rearrange of what's already there.  I'll take a look.